scripts/asr_language_modeling/ngram_lm/kenlm_utils.py

```diff
@@ -221,3 +224,18 @@ def write_dataset(chunks, path):
             for text in chunks[chunk_idx]:
                 line = ' '.join(text)
                 path.write((line + '\n').encode())
+
+
+def save_flashlight_lexicon(tokenizer, kenlm_file):
+    save_path = os.path.dirname(kenlm_file)
+    os.makedirs(save_path, exist_ok=True)
+
+    lex_file = os.path.join(save_path, os.path.splitext(os.path.basename(kenlm_file))[0] + '.flashlight_lexicon')
+
+    logging.info(f"Writing Lexicon file to: {lex_file}...")
+    with open(lex_file, "w", encoding='utf_8', newline='\n') as f:
+        lexicon = create_lexicon(tokenizer)
+        lexicon.pop("<unk>")
+        for word in lexicon:
+            f.write("{w}\t{s}\n".format(w=word, s=lexicon[word][0][0]))
```

Comment:
yes, but .format() is much faster

Reply:
Could you elaborate it? I found opposite information.

Reply:
F-strings are much more readable. Unless you do very tricky things requiring a lot of runtime, f-strings should be preferred.

Reply:
https://www.reddit.com/r/Python/comments/pivojb/performance_comparison_of_string_formatting/ claims that f-strings are faster. But anyway, it can vary between different Python versions. 
In most of our code, readability is more important. Use C++ if you need speed :)

Reply:
Agreed on fstring. The few seconds saved on millions of files is not much of an impact. Not unless we are in the sub 10 seconds processing range for writing the lexicon

Reply:
done